### PR TITLE
Max node per role under autoreg ,moved unreg on accounting stop on the connection profile and vlan filters enhancement

### PR DESCRIPTION
**Pull request by fdurand**
# Description
Allow PacketFence to test MAx device allowed by role when autoregister
Moved unregister on accounting stop parameter on the connection profile
Allow a value return by the vlan filters to be dynamic.

# Impacts
Radius

# Delete branch after merge
YES

# NEWS file entries

## Enhancements
* Allow max node per role on autoregister
* Moved unregister on accounting stop parameter on the connection profile
* Vlan filters can be set to ${node_info.category} and it will return the current category of the device

### CODE REVIEW

**Comment by julsemaan** [comment]
Need upgrade instructions for the unreg_on_accounting_stop parameter

File: html/pfappserver/lib/pfappserver/Form/Config/ProfileCommon.pm [hunk 1]
```diff
@@ -199,6 +199,24 @@ has_field 'autoregister' =>
...
+
+has_field 'unreg_on_acct_stop' =>
+  (
+   type => 'Toggle',
+   label => 'Automatically deregister devices on accounting stop',
+   checkbox_value => 'enabled',
+   unchecked_value => 'disabled',
+   default => 'disabled',
+   tags => { after_element => \&help,
+             help => 'This activates automatic deregistation of devices for the profile if PacketFence receive an radius accounting stop.' },
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/Form/Config/ProfileCommon.pm]: receives a RADIUS accounting stop for a device

File: lib/pf/Connection/Profile.pm [hunk 2]
```diff
@@ -609,6 +609,17 @@ sub canAccessRegistrationWhenRegistered {
...
 
+=item unregOnAcctStop
+
+Deregister device on accounting stop
+
+=cut
+
+sub unregOnAcctStop {
+    my ($self) = @_;
+    return $self->{'_unreg_on_acct_stop'};
```
Comment (julsemaan) [inline lib/pf/Connection/Profile.pm]: isenabled would be good here

File: lib/pf/role.pm [hunk 3]
```diff
@@ -415,7 +415,7 @@ sub getRegisteredRole {
     # the role based on the rules defined in the different authentication sources.
     # FIRST HIT MATCH
     elsif ( defined $args->{'user_name'} && $args->{'connection_type'} && ($args->{'connection_type'} & $EAP) == $EAP ) {
-        if ( isdisabled($profile->dot1xRecomputeRoleFromPortal) ) {
+        if ( isdisabled($profile->dot1xRecomputeRoleFromPortal)  || $args->{'autoreg'} == 1) {
```
Comment (julsemaan) [inline lib/pf/role.pm]: It seems autoreg can be set to 'yes' which is a string value and doesn't seem to be ever set to a number

**Comment by fdurand** [comment]
Bump

**Comment by extrafu** [comment]
@julsemaan Bump bump.

**Comment by julsemaan** [comment]
Still needs upgrade instructions for the unreg_on_accounting_stop parameter

**Pull request merged**